### PR TITLE
Let instructor/TAs skip comments when comparing

### LINKED ISSUE #534: Let instructor/TAs skip comments when comparing
Allow instructors/teaching assistants to skip adding comments for each answer when comparing.

Workaround for now would be to generate comments (default text 'Feedback') for each answer before the comparison process begins.

### PR DESCRIPTION
**Pull request by wynnset**
If instructor/TA is comparing:
- make feedback (answer comment) field optional AND
- remove the required asterisk from the interface

Fixes #534

### CODE REVIEW

**Comment by wynnset** [comment]
Ready for review @andrew-gardener - not sure if we want to go any further than this

File: compair/static/modules/comparison/comparison-form-partial.html [hunk 1]
```diff
@@ -151,22 +151,22 @@ <h3 class="text-center"><compair-logo type="'scale'"></compair-logo> Answer {{se
                 <fieldset>
                     <legend>Peer Feedback</legend>
                     <div class="col-md-6">
-                        <label class="cke-label required-star">
+                        <label class="cke-label {{commentRequired ? ' required-star' : ''}}">
```
Comment (andrew-gardener) [inline compair/static/modules/comparison/comparison-form-partial.html]: use `ng-class` for conditional classes

File: compair/static/modules/comparison/comparison-form-partial.html [hunk 2]
```diff
@@ -151,22 +151,22 @@ <h3 class="text-center"><compair-logo type="'scale'"></compair-logo> Answer {{se
...
                             Your private feedback for Answer {{firstAnsNum}}'s author
                             <i class="fa fa-arrow-down"></i>
                         </label>
                         <textarea
-                            ng-model="answer1.comment.content" rows="8" ckeditor="editor1Options" required>
+                            ng-model="answer1.comment.content" rows="8" ckeditor="editor1Options" ng-required="{{commentRequired}}">
```
Comment (andrew-gardener) [inline compair/static/modules/comparison/comparison-form-partial.html]: maybe just use `canManageAssignment` as the condition instead of `commentRequired`

**Comment by wynnset** [comment]
@andrew-gardener the issue is that the blank draft comments are saved when the compare page loads, before the user has even taken any action. I thought there was a purpose to this. Should I remove that functionality? Why are we currrently creating the entry for those comments when the page loads?

**Comment by andrew-gardener** [comment]
@wynnset Yes there is a purpose to creating the blank draft answer comments when it loads (it is for learning analytics, we need to have an id for the object to track it before its 'created' by the user).

You shouldn't need to change that part, just don't save the comment when the comparison is submitted if `canManageAssignment` and content is blank. If you send the blank comment to be saved, the backend will reject it but it would be better to not send it at all

**Comment by wynnset** [comment]
@andrew-gardener I've updated those. Also ran tests and it looks like it passed. We can wait for travis to verify.

**Pull request merged**